runtime/download_test.go

```diff
@@ -0,0 +1,46 @@
+package runtime
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"go4.org/xdgdir"
+)
+
+func Test_downloadFileWithCache(t *testing.T) {
```

Comment:
Optional suggestion for this test: I think the properties most important for this function are

- the file content matches what's on the server
- the GET request only occurs once

The filename returned isn't important to the caller and may change if we decide how and where we store the file.